inference-engine/src/mkldnn_plugin/nodes/mkldnn_roi_pooling_node.cpp

```diff
@@ -372,14 +446,15 @@ void MKLDNNROIPoolingNode::createPrimitive() {
         roi_pooling_kernel->create_ker();
 }
 
-void MKLDNNROIPoolingNode::execute(mkldnn::stream strm) {
+template<typename T>
+void MKLDNNROIPoolingNode::execute() {
```

Comment:
It looks like we can avoid using the template at all, since we only  use pointer manipulations here without type specific arithmetic. It will save the binary size. @dmitry-gorokhov what do you think about it?

Reply:
From my understanding, having conditional compilation feature leads to the following approach:
- if users do not care about binary size, those types of size optimizations are useless for them
- if users do care about binary size, they should be aware (and we should make them to be aware) of conditional compilation feature, so those types of size optimizations are useless for them as well.

Reply:
The problem here is that not for all users who cares about binary size conditional compilation is applicable solution. So we have to care about binary size in all use cases. Please provide information how library size is changing with this patch.

Reply:
+20KB

Reply:
Ok. Lets keep your impl